packages/cinterop/src/jvm/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -954,8 +954,41 @@ actual object RealmInterop {
         )
     }
 
-    actual fun realm_sync_client_config_new(): RealmSyncClientConfigurationPointer {
-        return LongPointerWrapper(realmc.realm_sync_client_config_new())
+    actual fun realm_sync_client_config_new(appId: String): RealmSyncClientConfigurationPointer {
+        val ptr: RealmSyncClientConfigurationPointer = LongPointerWrapper(realmc.realm_sync_client_config_new())
+        realmc.realm_sync_client_config_set_default_binding_thread_observer(
+            ptr.cptr(),
+            object : SyncThreadObserver {
+                override fun threadName(): String {
+                    return "SyncThread-$appId"
+                }
+
+                override fun onCreated() {
+                    // We cannot set the name on JNI side as it would require access to JNIEnv before
+                    // we attach it, so we set the thread name after it is created.
+                    Thread.currentThread().name = threadName()
+                }
+
+                override fun onDestroyed() {
+                    // Do nothing
+                    // Thread is destroyed in the JNI side
+                }
+
+                @Suppress("TooGenericExceptionThrown")
+                override fun onError(error: String) {
+                    // TODO Wait for https://github.com/realm/realm-core/issues/4194 to correctly
+                    //  log errors. For now, just throw an Error. Exceptions from the Sync Client
+                    //  indicate something is fundamentally is wrong on the Sync Thread.
+                    //
+                    //  As any exception here will be thrown out of context with no way to catch it, we
+                    //  turn it into an Error. Also, in Realm Java, these exceptions have only been
+                    //  reported when we integrated features ourselves, which indicates we want to
+                    //  make it very visible when they happen.
```

Comment:
Hmm. Seems like the PR description and the implementation is a bit schizophrenic about this ... which one do you prefer. If it is right that we shouldn't ever see this I guess I would go with throwing and updating the comment as suggested, but ... 🙈 
```suggestion
                    // TODO Wait for https://github.com/realm/realm-core/issues/4194 to correctly
                    //  log errors. For now, just throw an Error as exceptions from the Sync Client
                    //  indicate that something is fundamentally wrong on the Sync Thread. 
                    //  In Realm Java this has only been reported during development of new
                    //  features, so throwing an Error seems appropriate to increase visibility.
```